medmodels/data_synthesis/mtgan/model/critic/loss.py

```diff
@@ -0,0 +1,123 @@
+"""Loss function for Critic.
+
+WGANGPLoss: Wasserstein-Loss with gradient penalty for Critic
+"""
+
+from typing import Tuple
+
+import torch
+from torch import autograd, nn
+
+from medmodels.data_synthesis.mtgan.model.critic.critic import Critic
+
+
+class CriticLoss(nn.Module):
+    """Critic Loss: Wasserstein-Loss with gradient penalty for Critic."""
+
+    def __init__(self, critic: Critic, lambda_: float) -> None:
+        """Constructor for the Critic Wasserstein-Loss with gradient penalty.
+
+        Args:
+            critic (Critic): Critic model.
+            lambda_ (int): Gradient penalty coefficient.
+        """
+        super().__init__()
+        self.critic = critic
+        self.lambda_ = lambda_
```

Comment:
could there be a better name to use here instead of `lambda` so you don't have to add the `_`? If nothing comes to mind then that's fine :)